### PR TITLE
increase zeromq sockets

### PR DESCRIPTION
**Pull request by tuxology**
Fixes download dependency issue we sometimes face with large dependency list
- [x] Test on large repo

### CODE REVIEW

File: src/main/scala/ai/privado/utility/ConcurrentProcessor.scala [hunk 1]
```diff
@@ -12,7 +12,8 @@ import scala.concurrent.{Await, Future}
 
 object ConcurrentProcessor {
   val zContext = new ZContext()
-  val STOP     = "STOP"
+  zContext.getContext.setMaxSockets(4096)
```
Comment (pandurangpatil) [inline src/main/scala/ai/privado/utility/ConcurrentProcessor.scala:15]: @tuxology I guess we should take this value as an input to CLI by setting this value of 4096 as a default value to that input.

Comment (tuxology) [reply src/main/scala/ai/privado/utility/ConcurrentProcessor.scala]: Updated and tested this with YAML on pay repo. We do need to specific the value in string for now, eg "2048". Default has been set for 4096 and upper limit is 10,000

File: src/main/scala/ai/privado/utility/ConcurrentProcessor.scala [hunk 2]
```diff
@@ -39,13 +40,19 @@ object ConcurrentProcessor {
...
   private val logger = LoggerFactory.getLogger(this.getClass)
   // PUSH socket maintained for each task thread (Future)
   private val queue: ThreadLocal[ZMQ.Socket] = new ThreadLocal[ZMQ.Socket]
   // Unique ZeroMQ inproc:// topic used for each instance of ConcurrentProcessor.
   private val topic: String = UUID.randomUUID().toString
 
+  // set maximum sockets that can be used
+  private val count          = rules.getSystemConfigByKey(Constants.MAX_SOCKET_COUNT)
+  private val maxSocketCount = if (count.isEmpty) 1024 else if (count.toInt > 4000) 1024 else count.toInt
+  println(maxSocketCount)
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/utility/ConcurrentProcessor.scala]: @tuxology Can we have a little descriptive message along with socket count to understand this on console?

Comment (khemrajrathore) [reply src/main/scala/ai/privado/utility/ConcurrentProcessor.scala]: Nevermind was refering a older commit

**Comment by khemrajrathore** [comment]
@tuxology We will need a corresponding PR on privado with the default value being set there in systemConfig

**Review by pandurangpatil — APPROVED** [review]
looks good

**Pull request merged**